p2p/types/proto/p2p_config.proto

```diff
@@ -8,6 +8,8 @@ message P2PConfig {
   uint32 consensus_port = 2;
   bool use_rain_tree = 3;
   bool is_empty_connection_type = 4; // TODO (Drewsky) switch back to enum
+  bool use_rain_tree_redundancy_layer = 5;
+  bool use_rain_tree_cleanup_layer = 6;
 }
 
```

Comment:
I think we should have a separate proto for this:

```proto
message RainTreeConfig {
  rain_tree_redundancy_layer_enabled = 1;
  rain_tree_cleanup_layer_enabled = 2;
}
```

This way, we only need to pass in the raintree config into `NewRainTreeNetwork`, not the entire P2P config.